platforms/software/dependency-management/src/main/java/org/gradle/api/internal/artifacts/configurations/DefaultConfiguration.java

```diff
@@ -756,7 +758,7 @@ public ResolverResults call(BuildOperationContext context) {
                 try {
                     results = resolver.resolveGraph(DefaultConfiguration.this);
                 } catch (Exception e) {
-                    throw exceptionMapper.contextualize(e, DefaultConfiguration.this);
+                    throw exceptionMapper.mapFailure(e, "dependencies", displayName.getDisplayName());
```

Comment:
🤔 Can we align the field type here as well as the field name and method name here?  

If it's now going to be a mapper calling map that doesn't make any use of `ResolveException`, and that wraps exceptions instead of adding context to them, perhaps instead of `ResolveExceptionContextualizer` it should be called `ResolutionFailureHandler`/`Typer`/`Mapper`?

It could also be moved within `DefaultConfiguration` as an inner class, or at least be made package-protected.

Reply:
Yeah I've wanted to rename this to `ResolveExceptionMapper` for a while but have not just to avoid touching so many files. I've renamed it.

Reply:
I'm going to leave this in the same place as it is for now. If we move the code that detects if settings repositories are ignored, then that removes enough logic from this class that we should then delete it and move the remaining logic to `ResolutionHost` as default methods